[DEVCOMMS-1214] Feature/openFinance-choAPI (#1899)
mercadopago/devsite-docs: guides/checkout-api-v2/integrate-openfinance.es.md

# Open Finance
Open Finance es un nuevo sistema financiero que le permite tener un dominio total de sus datos y mayor libertad con sus finanzas. Con él, es posible compartir historias financieras con otras instituciones para acceder a productos y servicios más ventajosos.

Con Open Finance, se pueden hacer pagos Pix con saldos de otras cuentas bancarias, **en el Checkout API**.

> NOTE
> 
> Importante
> 

> > Para usar esta función, debe registrar una clave Pix en su cuenta de Mercado Pago. Si aún no lo ha hecho, [haga clic aquí](https://www.youtube.com/watch?v=60tApKYVnkA) para obtener más información sobre cómo registrarse.

Reviewer: ```suggestion
> Para usar esta función, debes registrar una clave Pix en tu cuenta de Mercado Pago. Si aún no lo has hecho, [haz clic aquí](https://www.youtube.com/watch?v=60tApKYVnkA) para obtener más información sobre cómo registrarse.
```

## Creación de un pago Pix con Open Finance
Para iniciar un pago utilizando Open Finance, debe tener configurado previamente el método de pago Pix. Para obtener instrucciones de configuración e integración, [consulte la documentación](/developers/es/docs/checkout-api/prerequisites).

> Una vez que haya configurado correctamente el método de pago, deberá agregar nueva información a la solicitud para [crear el pago](/developers/es/reference/payments/_payments/post), a través del parámetro `point_of_interaction`, que indica el modo de financiación abierta. Esto es válido tanto a través de la API como a través de nuestros SDK, como se muestra en los siguientes ejemplos:

Reviewer: ```suggestion
Una vez que hayas configurado correctamente el método de pago, deberás agregar nueva información a la solicitud para [crear el pago](/developers/es/reference/payments/_payments/post) a través del parámetro `point_of_interaction`, que indica el modo de financiación abierta. Esto es válido tanto a través de la API como a través de nuestros SDK, como se muestra en los siguientes ejemplos:
```

    }'
    ```
]]]

> WARNING
> 
> Informaciones importantes
> 
> - Para este tipo de método de pago, **el número de CPF es obligatorio**.
> - El parámetro `callback_url` tiene que ser la URL de la pantalla de feedback.

La respuesta de la solicitud será muy similar a la respuesta de un pago con un Pix común, con algunas modificaciones:

* El valor de **openfinance** en `point_of_interaction.linked_to`
* Valor **nulo** en `point_of_interaction.transaction_data.qr_code` y `point_of_interaction.transaction_data.qr_code_base64`
* Valor de `point_of_interaction.transaction_data.ticket_url` con URL a la aplicación que finaliza el pago por Open Finance

Respuesta

```json
{
  "point_of_interaction": {
    "linked_to": "openfinance",
    "transaction_data": {
      "qr_code": null,
      "ticket_url": "https://mercadopago.com.br/payments/12/openfinance?caller_id=1&hash=1111",
      "qr_code_base64": null
    }
  }
}
```

## Orientar al usuario para finalizar el pago
Con la respuesta de solicitud de creación de pago, simplemente use la URL disponible en el objeto `point_of_interaction.transaction_data.ticket_url` para que el comprador pueda finalizar el pago a través de Open Finance.

> Para esto, puede redirigir al usuario a la URL directamente o presentar un botón que redirija cuando se haga clic.

Reviewer: ```suggestion
Para esto, puedes redirigir al usuario a la URL directamente o presentar un botón que redirija cuando se haga clic.
```

La imagen siguiente muestra un ejemplo de este proceso en la vista del usuario:

![Payment flow sample](/images/api/api-integrate-openfinance.gif)

## Probando tu integración

> Para probar la integración, debe usar una credencial de prueba al crear un pago y tener un usuario de prueba con una clave Pix registrada. [Haga clic aquí](/developers/es/docs/checkout-api/integration-test/make-test-purchase) para acceder a la documentación y encontrar más detalles sobre cómo hacerlo.

Reviewer: ```suggestion
Para probar la integración, debes usar una credencial de prueba al crear un pago y tener un usuario de prueba con una clave Pix registrada. [Haz clic aquí](/developers/es/docs/checkout-api/integration-test/make-test-purchase) para acceder a la documentación y encontrar más detalles sobre cómo hacerlo.
```

> Cree un pago de Pix Open Finance y utilice la URL disponible en el atributo `point_of_interaction.transaction_data.ticket_url` para acceder a la página de redirección.

Reviewer: ```suggestion
Crea un pago de Pix Open Finance y utiliza la URL disponible en el atributo `point_of_interaction.transaction_data.ticket_url` para acceder a la página de redirección.
```

# Open Finance

> Open Finance es un nuevo sistema financiero que te permite tener un dominio total de tus datos y mayor libertad con tus finanzas. Con él, es posible compartir historias financieras con otras instituciones para acceder a productos y servicios más ventajosos.

Reviewer 2: ```suggestion

Open Finance es un nuevo sistema financiero que te permite tener un dominio total de tus datos y mayor libertad con tus finanzas. Con él, es posible compartir historias financieras con otras instituciones para acceder a productos y servicios más ventajosos.
```

    ```
    ```curl
    curl --location --request POST 'https://api.mercadopago.com/v1/payments' \
    --header 'Authorization: Bearer TOKEN' \
    --header 'Content-Type: application/json' \
    --data-raw '{
        "transaction_amount": 1000,
        "description": "Teste Pix Open Finance",
        "payment_method_id": "pix",
        "payer": {
            "email": "test_user_19734329@testuser.com"
        },
        "point_of_interaction": {
            "linked_to": "openfinance"
        },
        "callback_url": "https://example.com"
    }'
    ```
]]]

> WARNING
> 
> Atención
> 
> Para este tipo de medio de pago, **el número de CPF es obligatorio**.
> El parámetro `callback_url` tiene que ser la URL de la pantalla de feedback.

La respuesta de la solicitud será muy similar a la respuesta de un pago con un Pix común, con algunas modificaciones:

* El valor de **openfinance** en `point_of_interaction.linked_to`
* Valor **nulo** en `point_of_interaction.transaction_data.qr_code` y `point_of_interaction.transaction_data.qr_code_base64`
* Valor de `point_of_interaction.transaction_data.ticket_url` con URL a la aplicación que finaliza el pago por Open Finance

> Respuesta

Reviewer 2: ```suggestion
Respuesta:
```
